### PR TITLE
Implementation of ConfigParamRegistry::isSet()

### PR DESCRIPTION
**Pull request by pbuehler**
The isSet method allows to test whether an option was set on the command line.

See also "Using command line options in DataProcessorSpec" in  Framework/Core/COOKBOOK.md for more details.

### CODE REVIEW

File: Framework/Core/src/CommonDataProcessors.cxx [hunk 1]
```diff
@@ -216,38 +216,41 @@ DataProcessorSpec
...
-        filemode = fmo;
-      }
-      if (ntfm > 0) {
-        ntfmerge = ntfm;
+    std::string fnb, fmo;
+    int ntfm = 0;
+    if (ic.options().isSet("json-file")) {
+      auto fnjson = ic.options().get<std::string>("json-file");
+      if (!fnjson.empty()) {
+        tie(fnb, fmo, ntfm) = dod->readJson(fnjson);
```
Comment (ktf) [inline Framework/Core/src/CommonDataProcessors.cxx:224]: `auto [fnb, fmo, ntfm] = `

File: Framework/Core/src/CommonDataProcessors.cxx [hunk 2]
```diff
@@ -359,11 +362,11 @@ DataProcessorSpec
...
-    {{"json-file", VariantType::String, "", {"Name of the json configuration file"}},
-     {"res-file", VariantType::String, "", {"Default name of the output file"}},
-     {"res-mode", VariantType::String, "", {"Creation mode of the result files: NEW, CREATE, RECREATE, UPDATE"}},
-     {"ntfmerge", VariantType::Int, -1, {"Number of time frames to merge into one file"}},
-     {"keep", VariantType::String, "", {"Comma separated list of ORIGIN/DESCRIPTION/SUBSPECIFICATION:treename:col1/col2/..:filename"}}}};
+    {{"json-file", VariantType::String, {"Name of the json configuration file"}},
+     {"res-file", VariantType::String, {"Default name of the output file"}},
+     {"res-mode", VariantType::String, {"Creation mode of the result files: NEW, CREATE, RECREATE, UPDATE"}},
+     {"ntfmerge", VariantType::Int, {"Number of time frames to merge into one file"}},
+     {"keep", VariantType::String, {"Comma separated list of ORIGIN/DESCRIPTION/SUBSPECIFICATION:treename:col1/col2/..:filename"}}}};
```
Comment (ktf) [inline Framework/Core/src/CommonDataProcessors.cxx:369]: I still think we should have sensible defaults for these. Isn't the point of this feature to actually allow for those?

File: Framework/Core/src/CommonDataProcessors.cxx (hunk 1 above)
Comment (pbuehler) [reply Framework/Core/src/CommonDataProcessors.cxx:224]: true, have removed auto for some reason earlier, but can put it back again.

File: Framework/Core/src/CommonDataProcessors.cxx (hunk 2 above)
Comment (pbuehler) [reply Framework/Core/src/CommonDataProcessors.cxx:369]: This is indeed a minimal improvement with respect to the current situation.

Currently options can be defined with or without default value. There is no way to test whether an option has been set or not. The access to an option defined with default value always returns a valid value - either the value set on the command line or the default value. However, if one accesses an option which was defined without default value and the option is not set on the command line, then the returned value is undefined, respectively the program crashes. So the only save method is to use options with default values, and for these there is no method to test whether it has been set or not.

With this modification it is now possible to test an option which was defined without default value, whether it has been set or not. The same test method applied to options with default value returns in any case true.

This by the way mimics the behaviour of the boost::program_options, which also allows to defined options with and without default values.

In a next step I can try to implement the isSet method such, that also options with default values can be properly tested.

**Comment by ktf** [comment]
Thank you for your explanation. Merging.

**Pull request merged**